staging/src/k8s.io/kubectl/pkg/polymorphichelpers/helpers.go

```diff
@@ -71,13 +76,27 @@ func GetFirstPod(client coreclient.PodsGetter, namespace string, selector string
 	defer cancel()
 	event, err := watchtools.UntilWithoutRetry(ctx, w, condition)
 	if err != nil {
-		return nil, 0, err
+		return nil, err
 	}
-	pod, ok := event.Object.(*corev1.Pod)
+
+	_, ok := event.Object.(*corev1.Pod)
 	if !ok {
-		return nil, 0, fmt.Errorf("%#v is not a pod event", event)
+		return nil, fmt.Errorf("%#v is not a pod event", event)
+	}
+	return podList, nil
```

Comment:
We either need to add the pod we waited for to `podList` or return a new list containing it.

Reply:
The second half of the function is for if the initial pod list was empty. [Here](https://github.com/kubernetes/kubernetes/blob/e798b9c269627a10291f36a88b4fa13f034ab57f/staging/src/k8s.io/kubectl/pkg/polymorphichelpers/helpers.go#L53-L56) we were returning if the list had any entries.


[Here](https://github.com/kubernetes/kubernetes/blob/e798b9c269627a10291f36a88b4fa13f034ab57f/staging/src/k8s.io/kubectl/pkg/polymorphichelpers/helpers.go#L72-L80) we were watching to see if any pods are spun up and then return the first pod that is observed.

Does my reading of that make sense?

Reply:
ahh yes i finally get it, in the context of GetFirstPod this totally makes sense as like a last try